contributors/design-proposals/encryption.md

```diff
@@ -0,0 +1,204 @@
+# Encryption
+
+## Abstract
+
+The scope of this proposal is to ensure that resources can be encrypted at the datastore layer. Encrypting data at REST and via third party vendors is a desired feature but outside the scope of this proposal. Encryption will be optional for any resource but we suspect it will be used for the Secret resource in most cases.
+
+Allowing sensitive data to be encrypted adheres to best practices as well as other requirements such as HIPAA.
+
+How encryption keys are delivered to the machine running the Kubernetes apiserver is of relevance - we assume that the encryption at rest pattern is loosely coupled to how those keys are delivered and secured on disk.
+
+In general, full disk encryption of the volumes storing etcd data is generally preferred - this proposal focuses on scenarios where additional protection is desired against malicious parties gaining read access to the etcd API or a running etcd instance without access to memory of the etcd process.
+
+## High level design
+Before a resource is written to etcd and before it is read, an encryption provider will take the plaintext data and encrypt it.
+These providers will be able to be created and turned on depending on the users needs or requirements and will adhere to an encryption interface.
+This interface will provide the abstraction to allow various encryption mechanisms to be implemented, as well as for the method of encryption to be rotated over time.
+It should be possible for the distribution of keys to the apiserver to be separated out (injected in by a higher level security process) or to be directly
+requested by the provider implementation. For the first iteration, a default provider will be developed and will run as part of the kube-apiserver.
```

Comment:
+1, lets resolve how the key is going to get delivered for the default implementation. Are we only supporting a file on disk? What do we do if it is missing or the wrong key? Should we support passing it in "manually" somehow? If so, will the cluster still work while we're waiting on the key?